extensions/common/iam/oauth2/oauth2-daps/src/main/java/org/eclipse/edc/iam/oauth2/daps/DapsValidationRule.java

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.iam.oauth2.daps;
+
+import org.eclipse.edc.jwt.spi.TokenValidationRule;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.iam.ClaimToken;
+import org.eclipse.edc.spi.result.Result;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class DapsValidationRule implements TokenValidationRule {
+
+    private final boolean validateReferring;
+    private final boolean validateIssuer;
+
+    public DapsValidationRule(boolean validateReferring, boolean validateIssuer) {
+        this.validateReferring = validateReferring;
+        this.validateIssuer = validateIssuer;
+    }
+
+    @Override
+    public Result<Void> checkRule(@NotNull ClaimToken toVerify, @Nullable Map<String, Object> additional) {
+        if (additional != null) {
+            var issuerConnector = getString(additional, "issuerConnector");
+            if (validateIssuer && issuerConnector == null) {
+                return Result.failure("Required issuerConnector is missing in message");
+            }
+
+            var securityProfile = getString(additional, "securityProfile");
+
+            return verifyTokenIds(toVerify, issuerConnector, securityProfile);
+
+        } else {
+            throw new EdcException("Missing required additional information for DAPS token validation");
+        }
+    }
+
+    private Result<Void> verifyTokenIds(ClaimToken jwt, String issuerConnector, @Nullable String securityProfile) {
+        //referringConnector (DAT, optional) vs issuerConnector (Message-Header, mandatory)
+        var referringConnector = jwt.getClaim("referringConnector");
+
+        if (validateReferring && !issuerConnector.equals(referringConnector)) {
+            return Result.failure("referringConnector in token does not match issuerConnector in message");
+        }
+
+        //securityProfile (DAT, mandatory) vs securityProfile (Message-Payload, optional)
+        try {
+            var tokenSecurityProfile = jwt.getClaim("securityProfile");
+
+            if (securityProfile != null && !securityProfile.equals(tokenSecurityProfile)) {
+                return Result.failure("securityProfile in token does not match securityProfile in payload");
```

Comment:
Nit we should begin messages with capitalization, e.g. "The claim `securityProfile`.....